roles/openshift_master_facts/tasks/main.yml

```diff
@@ -73,3 +73,9 @@
       oauth_templates: "{{ openshift_master_oauth_templates | default(None) }}"
       oauth_always_show_provider_selection: "{{ openshift_master_oauth_always_show_provider_selection | default(None) }}"
       image_policy_config: "{{ openshift_master_image_policy_config | default(None) }}"
+      builddefaults_http_proxy: "{{ openshift_builddefaults_http_proxy | default(None) }}"
+      builddefaults_https_proxy: "{{ openshift_builddefaults_https_proxy | default(None) }}"
+      builddefaults_no_proxy: "{{ openshift_builddefaults_no_proxy | default(None) }}"
+      builddefaults_git_http_proxy: "{{ openshift_builddefaults_git_http_proxy | default(None) }}"
+      builddefaults_git_https_proxy: "{{ openshift_builddefaults_git_https_proxy | default(None) }}"
+      builddefaults_config: "{{ openshift_builddefaults_json | default(builddefaults_yaml) }}"
```

Comment:
+1 Could use something like this to grab and set all of the `builddefaults` facts ie. `openshift.builddefaults.http_proxy` once the role has been added to `openshift_facts`.

```
- name: Set builddefaults facts
  openshift_facts:
    role: builddefaults
    openshift_env: "{{ hostvars[inventory_hostname]
                       | oo_merge_dicts(hostvars)
                       | oo_openshift_env }}" 
    openshift_env_structures:
    - 'openshift.builddefaults.*' 
```